blog/models.py

```diff
@@ -179,3 +173,25 @@ def _validate_recommended_articles_uniqueness(self) -> None:
                 raise ValidationError(message=f"'{stream_child.value}' is listed more than once!")
             else:
                 article_pages_set.add(stream_child.value)
+
+
+@receiver(post_save, sender=BlogArticlePage)
+def ensure_main_article_exists(sender: Any, instance: Any, **_kwargs: Dict[Any, Any]) -> None:
+    if not instance.is_main_article and not sender.objects.exclude(id=instance.id).filter(is_main_article=True):
```

Comment:
Please call `exists()` method on query -> 
`not sender.objects.exclude(id=instance.id).filter(is_main_article=True).exists()`
It will give you True/False value

Reply:
Oh, I've forgotten about it. I believe I saw this being omitted somewhere in some template as well, so I probably based this code on that. Maybe I'll use this opportunity to fix those as well.

Reply:
Done.